### PR TITLE
Implement MessageSourceAware where missing

### PR DESCRIPTION
**Pull request by amergey**
Closes gh-8951

### CODE REVIEW

File: core/src/main/java/org/springframework/security/authentication/AbstractUserDetailsReactiveAuthenticationManager.java [hunk 1]
```diff
@@ -164,6 +167,11 @@ public void setPostAuthenticationChecks(UserDetailsChecker postAuthenticationChe
 		this.postAuthenticationChecks = postAuthenticationChecks;
 	}
 
+	@Override
```
Comment (jzheaux) [inline core/src/main/java/org/springframework/security/authentication/AbstractUserDetailsReactiveAuthenticationManager.java:170]: Will you please add JavaDoc for these setters? In the JavaDoc, please add `@since 5.5`.

File: core/src/main/java/org/springframework/security/authentication/AbstractUserDetailsReactiveAuthenticationManager.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
 import reactor.core.scheduler.Scheduler;
 import reactor.core.scheduler.Schedulers;
 
+import org.springframework.context.MessageSource;
```
Comment (jzheaux) [inline core/src/main/java/org/springframework/security/authentication/AbstractUserDetailsReactiveAuthenticationManager.java:25]: Please make sure that each file uses the most recent copyright year. In this case, that would mean changing `2002-2019` to `2002-2020`.

File: core/src/main/java/org/springframework/security/authentication/AbstractUserDetailsReactiveAuthenticationManager.java [hunk 3]
```diff
@@ -164,6 +167,11 @@ public void setPostAuthenticationChecks(UserDetailsChecker postAuthenticationChe
 		this.postAuthenticationChecks = postAuthenticationChecks;
 	}
 
+	@Override
+	public void setMessageSource(MessageSource messageSource) {
+		this.messages = new MessageSourceAccessor(messageSource);
```
Comment (jzheaux) [inline core/src/main/java/org/springframework/security/authentication/AbstractUserDetailsReactiveAuthenticationManager.java:172]: It would also be nice if the setters ensured a non-null parameter, like with `Assert.notNull`.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks for the PR, @amergey! I've left some feedback inline.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks for the updates, @amergey. Lastly, would you please add tests that confirm that setting the message source has the desired effect on each class?

**Comment by amergey** [comment]
> Thanks for the updates, @amergey. Lastly, would you please add tests that confirm that setting the message source has the desired effect on each class?

@jzheaux I do not see what to test here, the bug was these classes were not implementing MessageSourceAware as they should, fixing the bug is juts to implement it and I do not see what relevant test we could add there. 
( MessageSourceAware behavior is already covered by tests like AutowireBeanFactoryObjectPostProcessorTests. )

**Comment by jzheaux** [comment]
> the bug was these classes were not implementing MessageSourceAware as they should, fixing the bug is juts to implement it

Good points. It's quite common in Spring Security to test that setters have non-null guards. For example, you can see this in `JdbcDaoImplTests#setMessageSourceWhenNullThenThrowsException`. I believe this PR should at least test that the setters check for `null`.

Additionally, I was thinking that tests would confirm that the `MessageSource` used by the class is the one that the setter provided. But, after trying it out myself on `SubjectDnX509PrincipalExtractorTests`, it seems to create a test that heavily overlaps other tests. Also, this kind of test isn't done anywhere else in the repo, so maybe we can leave those out.

Can you add tests that confirm that the setters check for null?

**Comment by amergey** [comment]
> Can you add tests that confirm that the setters check for null?
sure ! I have added 2 tests per class similar to what is done elsewhere

**Comment by jzheaux** [comment]
@amergey Thanks for the update! 

Sorry, I feel like I miscommunicated on my feedback - I'd prefer to not expose fields for the sake of testing. I think it's okay to only add tests that confirm that the setters check for null.

Do you feel strongly that the second of two tests in each class should stay? If so, let's use default scope instead of `protected` scope. If not, let's remove the tests.

**Comment by amergey** [comment]
> @amergey Thanks for the update!
> 
> Sorry, I feel like I miscommunicated on my feedback - I'd prefer to not expose fields for the sake of testing. I think it's okay to only add tests that confirm that the setters check for null.
> 
> Do you feel strongly that the second of two tests in each class should stay? If so, let's use default scope instead of `protected` scope. If not, let's remove the tests.

I just did that like this to be consistent with what existed. If you think it is better without the second test, then I can remove them and move scope to private

**Comment by jzheaux** [comment]
> I just did that like this to be consistent with what existed.

Thanks, @amergey. Can you humor me for a moment and point out where Spring Security is testing setters by making the field `protected`? I want to make sure I'm not missing something. I agree that the first of the two tests is consistent with the rest of Spring Security, but I'm having trouble seeing the second of the two tests as being consistent with what exists.

**Comment by amergey** [comment]
> > I just did that like this to be consistent with what existed.
> 
> Thanks, @amergey. Can you humor me for a moment and point out where Spring Security is testing setters by making the field `protected`? I want to make sure I'm not missing something. I agree that the first of the two tests is consistent with the rest of Spring Security, but I'm having trouble seeing the second of the two tests as being consistent with what exists.

When I said consistent I did not have in mind testing setters by making the field protected,  it was about having the "message" field itself protected (which seems to be protected when a class implements MessageSourceAware). 
I have seen the second test in another MessageSourceAware implementation, but I need to find which one I used as sample

**Review by jzheaux — APPROVED** [review]

**Comment by jzheaux** [comment]
I see, @amergey, thanks for clarifying.

This is now merged into `master`.
